### DISCUSSION TITLE
Holder-initiated revocation request

### DISCUSSION THREAD

**Discussion by vanhoanHoang** [walt-id/.github#50] [Ideas / Feature Requests]
Thank you the team for having worked hard to release the revocation feature. Actually, the revocation can be made via API by both user and issuer. 
I come with the idea of asking user to prove the key possession to be able to revoke a VC. Is it possible to integrate it directly in the Wallet ? 

Thank
Hoan Hoang

**Comment by taminobaumann** [comment]
Hey @vanhoanHoang, 

thanks for reaching out, and we are glad to hear you like our tools! 

In the near future, we will add authentication to the wallet-kit and ssi-kit APIs. This will ensure that only authorized users are able to access the functions provided by the API, including the ability to revoke credentials. Depending on the user's preference, either the user, the issuer, or both, will be able to revoke credentials.

At this time, we do not have plans to add business logic that allows users to do their own revocations. If we receive more requests for this feature, we will consider adding it. In the meantime, you can use our open-source tooling and extend it to include the desired functionality.

Hope it helps!
Best
Tamino

**Comment by waltkb** [comment]
Hi, the revocation supports two status entries: StatusList2021 and SimpleStatus2022.

StatusList2021 works by hosting an bitset over HTTP.

SimpleStatus2022 works by having an base (revocation) token, and an derived (revocation check) token. So the token in the VC is not the token that is able to revoke the VC.
